tests/suites/cloud/suite.js

```diff
@@ -0,0 +1,248 @@
+      os: new BalenaOS(
+        {
+          deviceType: this.suite.deviceType.slug,
+          network: this.suite.options.balenaOS.network,
+        },
+        this.getLogger()
+      ),
+    });
+
+    // unpack OS
+    await this.context.get().os.fetch();
+
+    await this.context.get().os.readOsRelease();
+
+    // get config.json for application
+    let config = await this.context
+      .get()
+      .cloud.balena.models.os.getConfig(this.context.get().balena.application, {
+        version: this.context.get().os.contract.version,
+      });
+
+    config.uuid = this.context.get().balena.uuid;
+
+    //register the device with the application, add the api key to the config.json
+    let deviceApiKey = await this.context
+      .get()
+      .cloud.balena.models.device.register(
+        this.context.get().balena.application,
+        this.context.get().balena.uuid
+      );
+    config.deviceApiKey = deviceApiKey.api_key;
+
+    // get newly registered device id, add to config.json
+    await Bluebird.delay(1000 * 10);
+    let devId = await this.context
+      .get()
+      .cloud.balena.models.device.get(this.context.get().balena.uuid);
+    config.deviceId = devId.id;
+    config.persistentLogging = true;
+
+    // get ready to populate DUT image config.json with the attributes we just generated
+    this.context.get().os.addCloudConfig(config);
+
+    // Teardown the worker when the tests end
+    this.suite.teardown.register(() => {
+      this.log("Worker teardown");
+      return this.context.get().worker.teardown();
+    });
+
+    // preload image with the single container application
+    this.log(`Device uuid should be ${this.context.get().balena.uuid}`)
+    this.log("Preloading image...");
+    await this.context.get().os.configure();
+    await this.context.get().cli.preload(this.context.get().os.image.path, {
+      app: this.context.get().balena.application,
+      commit: initialCommit,
+      pin: true,
+    });
+
+    this.log("Setting up worker");
+    await this.context
+      .get()
+      .worker.network(this.suite.options.balenaOS.network);
+
+    await this.context.get().worker.off();
+    await this.context.get().worker.flash(this.context.get().os.image.path);
+    await this.context.get().worker.on();
+
+    this.log("Waiting for device to be reachable");
+    await this.context.get().utils.waitUntil(async() => {
+      return await this.context
+        .get()
+        .cloud.balena.models.device.isOnline(this.context.get().balena.uuid) === true;
+    });
+
+    // Retrieving journalctl logs
+    this.suite.teardown.register(async () => {
+      await this.context.get().worker.archiveLogs(this.id, this.context.get().link);
+    });
+
+    this.log("Device is online and provisioned successfully");
+    await this.context.get().utils.waitUntil(async () => {
+      this.log("Trying to ssh into device");
+      let hostname = await this.context
+      .get()
+      .cloud.executeCommandInHostOS(
```

Comment:
Is this actually a cloud function?

Reply:
@klutchell it is in a way. It ssh's into the DUT via the proxyL https://github.com/balena-os/leviathan/blob/master/core/lib/components/balena/sdk.js#L86

Reply:
We could just SSH in directly like we do in the unmanaged suite if this isn't suitable though

Reply:
Nope, this looks good. Thanks for pointing me to the code!